server/game/gameSystems/DamageSystem.ts

```diff
@@ -23,7 +22,16 @@ export class DamageSystem<TContext extends AbilityContext = AbilityContext> exte
     protected override readonly targetTypeFilter = [WildcardCardType.Unit, CardType.Base];
 
     public eventHandler(event): void {
-        event.card.addDamage(event.damage);
+        switch (typeof event.damage) {
```

Comment:
I just realized there's an issue here - we need it to be the case that the `damage` value is set when the event is created, not when it is resolved.

There are two problems that could happen currently:

1. It is possible that attributes of the card could change during resolution, so that the `event.damage` function gives different values depending on when it is called
2. There are some cards such as Boba Fett's Armor that modify the damage value before it resolves, so the damage must be determined already

Maybe the simplest way to fix this is to do the calculation in `addPropertiesToEvent()` instead of here. That way it will be set once and not changed.